app/code/Magento/Catalog/Controller/Adminhtml/Product/Save.php

```diff
@@ -117,6 +117,7 @@ public function execute()
                 $productId = $product->getEntityId();
                 $productAttributeSetId = $product->getAttributeSetId();
                 $productTypeId = $product->getTypeId();
+                $this->getRequest()->setParam('type', $productTypeId);
```

Comment:
in did, but type is changed here so why incorrect param should still exists in param? this is not inceasing complexity but fixing the value in the request object as it is modified in product.